src/ray/gcs/redis_context.cc

```diff
@@ -620,6 +644,14 @@ Status RedisContext::Connect(const std::string &address,
   }
 }
 
+Status RedisContext::Reconnect(RedisConnectType type) {
```

Comment:
We can just always reconnect both since they connect to the same server.

Reply:
I don’t recommend doing it this way because, even though the server is the same, the connections are different. Honestly, I think the function calls here are a bit messy, and maybe we could refactor them in the future. The sync context and async context seem to interfere with each other. Initially, I tried reconnecting them together, but the program got stuck. I didn’t trace the function calls in detail, but when they are separated, it doesn’t get stuck.

However, if you still think it's necessary to reconnect the sync and async contexts together, I can try again and look into why it gets stuck.

Reply:
I think it's better to just reconnect both so that we can make sure both are connected to the same host. Otherwise, sync may connect to a new master while async may still connect to the old master. It's just easier to reason about if we reconnect both.